docs/HyperIndex/Advanced/loaders.md

```diff
@@ -0,0 +1,100 @@
+---
+id: loaders
+title: Loaders
+sidebar_label: Loaders
+slug: /loaders
+---
+
+## Overview
+
+Loaders provide an efficient way to define and load entities from the database, optimizing performance by minimizing I/O operations. They also enable you to load arrays of entities based on specific field values.
+
+Loaders are designed to reduce I/O, which is often the primary performance bottleneck in an indexer. By using loaders, you can group multiple requests into a single database round trip before processing a batch of events. The retrieved values are cached in memory, allowing the batch to be processed entirely in memory.
+
+## Example: Refactoring a Simple Handler
+
+Let's explore how to convert a basic handler to use a loader. Below is an example of a handler that loads values at runtime, on demand. Note: Built-in caching still ensures that reloading the same value multiple times in a batch doesn't result in multiple database hits.
```

Comment:
I assume the note might create confusion
```suggestion
Let's explore how to convert a basic handler to use a loader. Below is an example of a handler that loads values at runtime, on demand.
```

Reply:
Ok cool, I just wanted to create that note so that people understand the inline loader will still cache entities and not create extra roundtrips when it's not necessary.